.github/workflows/gh-build-and-test.yml

```diff
@@ -1,126 +1,323 @@
+    # The build stage could fail but we want the CI to keep moving.
+    if: ${{ (github.repository_owner == 'nvidia') && always() }}
     permissions:
       id-token: write # This is required for configure-aws-credentials
       contents: read  # This is required for actions/checkout
-    runs-on: ${{ (inputs.host-platform == 'linux-x64' && 'linux-amd64-gpu-v100-latest-1') ||
-                 (inputs.host-platform == 'linux-aarch64' && 'linux-arm64-gpu-a100-latest-1') }}
+    runs-on: ${{ (matrix.runner == 'default' && matrix.host-platform == 'linux-64' && 'linux-amd64-gpu-v100-latest-1') ||
+                 (matrix.runner == 'default' && matrix.host-platform == 'linux-aarch64' && 'linux-arm64-gpu-a100-latest-1') ||
+                 (matrix.runner == 'H100' && 'linux-amd64-gpu-h100-latest-1-testing') }}
     # Our self-hosted runners require a container
     # TODO: use a different (nvidia?) container
     container:
-      options: -u root --security-opt seccomp=unconfined --privileged --shm-size 16g
+      options: -u root --security-opt seccomp=unconfined --shm-size 16g
       image: ubuntu:22.04
       env:
         NVIDIA_VISIBLE_DEVICES: ${{ env.NVIDIA_VISIBLE_DEVICES }}
     needs:
       - build
     steps:
+      - name: Run nvidia-smi to make sure GPU is working
+        shell: bash --noprofile --norc -xeuo pipefail {0}
+        run: nvidia-smi
+
       - name: Checkout ${{ github.event.repository.name }}
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      # TODO: we probably don't need this?
-      # - name: Setup
-      #   if: ${{ inputs.has-built != 'true' }}
-      #   uses: ./.github/actions/setup
-      #   with:
-      #     client-repo: ${{ github.event.repository.name }}
-      #     build-type: ${{ inputs.build-type }}
-      #     target-device: "${{ inputs.target-device }}"
-      #     host-platform: ${{ inputs.host-platform }}
-      #     build-mode: ${{ inputs.build-mode }}
-      #     upload-enabled: ${{ inputs.upload-enabled }}
-      #     python-version: ${{ inputs.python-version }}
-
-      - name: Call test action
-        uses: ./.github/actions/test
+      - name: Set environment variables
+        shell: bash --noprofile --norc -xeuo pipefail {0}
+        run: |
+          PYTHON_VERSION_FORMATTED=$(echo '${{ matrix.python-version }}' | tr -d '.')
+          if [[ "${{ matrix.host-platform }}" == linux* ]]; then
+            REPO_DIR=$(pwd)
+          elif [[ "${{ matrix.host-platform }}" == win* ]]; then
+            PWD=$(pwd)
+            REPO_DIR=$(cygpath -w $PWD)
+          fi
+
+          BUILD_CUDA_MAJOR="$(cut -d '.' -f 1 <<< ${{ needs.build.outputs.BUILD_CTK_VER }})"
+          TEST_CUDA_MAJOR="$(cut -d '.' -f 1 <<< ${{ matrix.cuda-version }})"
+          if [[ $BUILD_CUDA_MAJOR -gt $TEST_CUDA_MAJOR ]]; then
+            SKIP_CUDA_BINDINGS_TEST=1
+          else
+            SKIP_CUDA_BINDINGS_TEST=0
+          fi
```

Comment:
We cannot run `cuda.bindings` tests when using `cuda.bindings` 11.x with CTK 12.x